src/plugins/vis_type_tagcloud/public/components/tag_cloud.tsx

```diff
@@ -20,61 +20,105 @@
 import d3 from 'd3';
 import d3TagCloud from 'd3-cloud';
 import { EventEmitter } from 'events';
+import { D3ScalingFunction } from '../types';
 
-const ORIENTATIONS = {
+const ORIENTATIONS: any = {
   single: () => 0,
-  'right angled': tag => {
+  'right angled': (tag: any) => {
     return hashWithinRange(tag.text, 2) * 90;
   },
-  multiple: tag => {
-    return hashWithinRange(tag.text, 12) * 15 - 90; //fan out 12 * 15 degrees over top-right and bottom-right quadrant (=-90 deg offset)
+  multiple: (tag: any) => {
+    return hashWithinRange(tag.text, 12) * 15 - 90; // fan out 12 * 15 degrees over top-right and bottom-right quadrant (=-90 deg offset)
   },
 };
-const D3_SCALING_FUNCTIONS = {
+const D3_SCALING_FUNCTIONS: D3ScalingFunction = {
   linear: () => d3.scale.linear(),
   log: () => d3.scale.log(),
   'square root': () => d3.scale.sqrt(),
 };
 
 export class TagCloud extends EventEmitter {
-  constructor(domNode, colorScale) {
+  _element: HTMLElement;
+  _d3SvgContainer: any;
```

Comment:
This can be typed strictly

Reply:
Have issues with setting type on `_d3SvgContainer`